babel/dates.py

```diff
@@ -943,11 +943,16 @@ def format_timedelta(
 
     def _iter_patterns(a_unit):
         if add_direction:
-            unit_rel_patterns = locale._data['date_fields'][a_unit]
-            if seconds >= 0:
-                yield unit_rel_patterns['future']
-            else:
-                yield unit_rel_patterns['past']
+            for key in (f"{a_unit}-{format}", a_unit):
+                unit_rel_patterns = locale._data['date_fields'].get(key)
+                if not unit_rel_patterns:
+                    continue
```

Comment:
I think we can simplify this to avoid the loop:

```suggestion
            fallback = locale._data['date_fields'][a_unit]
            unit_rel_patterns = locale._data['date_fields'].get(f"{a_unit}-{format}", fallback)
```

Reply:
Fair point! I pushed a version of this that avoids accessing the fallback if we do find the requested variant :)